### PR TITLE
Only category admin can manage choices #95

### LINKED ISSUE #95: Only category admin should be able to manage choices
Add in the authorization stuff to only allow category admins to update choices.

### PR DESCRIPTION
**Pull request by grimm**
Fixes #95

### CODE REVIEW

**Comment by grimm** [comment]
@gina-alaska/awrvi Ready for review.

**Comment by smacfarlane** [comment]
There should be tests for the ability model asserting and refuting different users capabilities around the choices.  See #93 for an example

**Comment by grimm** [comment]
@gina-alaska/awrvi 
Ready for review.

File: test/models/ability_test.rb [hunk 1]
```diff
@@ -16,4 +16,27 @@ def test_admins_can_manage_all_users
 
     assert ability.can?(:manage, User.new), 'User Admin cannot manage users'
   end
+
+  def test_users_cannot_create_choices
+    user = users(:one)
+    ability = Ability.new(user)
+
+    assert ability.cannot?(:create, Choice.new), "User can create a choice"
```
Comment (teknofire) [inline test/models/ability_test.rb]: Need to make sure normal user cannot :read, :create or :update Choices

File: test/models/ability_test.rb [hunk 2]
```diff
@@ -16,4 +16,27 @@ def test_admins_can_manage_all_users
...
+  end
+
+  def test_guests_can_only_read_choices
+    user = User.new
+    ability = Ability.new(user)
+
+    assert ability.can?(:read, choices(:med)), "Guest cannot read choice"
```
Comment (teknofire) [inline test/models/ability_test.rb]: Again guest user like normal user should not be able to read, create, update or delete.

File: test/models/ability_test.rb [hunk 3]
```diff
@@ -16,4 +16,27 @@ def test_admins_can_manage_all_users
...
+    assert ability.cannot?(:create, Choice.new), "Guest can create choice"
+  end
+
+  def test_only_category_admin_can_edit
+    user = users(:category_admin)
+    ability = Ability.new(user)
+
+    assert ability.can?(:manage, choices(:med)), "Category admin cannot manage choices"
+    assert ability.can?(:edit, choices(:low)), "category_admin cannot edit choices"
```
Comment (teknofire) [inline test/models/ability_test.rb]: Not sure why we have both of these, manage should be have this, unless there is some specific reason to call out the edit permission like they might not have :manage, but have :edit in the ability file.

**Comment by grimm** [comment]
Ready for review.  Looks like we need to change a rubocop rule?

**Comment by grimm** [comment]
Ready for review.

**Comment by teknofire** [comment]
We should make sure that we have the appropriate CRUD permission tests for each of those items.  Several of the tests are not dealing with delete permissions at all

**Comment by teknofire** [comment]
Ok, I think we need to take a quick step back.  There are lots of gaps in theses tests right now concerning the index permissions. 

We should be writing these from the point of the different actors and what they can or cannot do.  Not separate tests for each permission type.  So currently there should probably be 3 methods in the ability tests that cover the appropriate actions for `:create, :read, :update, :delete`.  I would also say get rid of the `:manage` permission checks as the handling all the `CRUD` permission should cover that.

Let's make sure to cover these scenarios
1. guests  (logged out users)
2. logged in users who do not own the index
3. logged in users who do own the index

Eventually this would need to include any specialized action like the finalized stuff as well.

**Comment by grimm** [comment]
Ok, but that made those methods really large so I had to up the AbcSize metric to 40 in order to have rubocop accept it.

**Comment by teknofire** [comment]
Much happier with this now, all the ability options should now be covered by the ability test.

:+1:

**Comment by smacfarlane** [comment]
:+1:

**Pull request merged**